File: lib_core/src/main/java/com/eelly/core/widget/refresh/api/IRefreshContent.java
package com.eelly.core.widget.refresh.api;

import android.animation.ValueAnimator.AnimatorUpdateListener;
import android.view.MotionEvent;
import android.view.View;
import android.view.ViewGroup;

/**
 * 刷新内容组件
 * @author Vurtne on 14-Dec-17
 */
public interface IRefreshContent {
    void moveSpinner(int spinner);
    boolean canRefresh();
    boolean canLoadMore();
    int getMeasuredWidth();
    int getMeasuredHeight();
    void measure(int widthSpec, int heightSpec);
    void layout(int left, int top, int right, int bottom);

    View getView();
    View getScrollableView();
    ViewGroup.LayoutParams getLayoutParams();

    void onActionDown(MotionEvent e);
    void onActionUpOrCancel();

    void fling(int velocity);
    void setUpComponent(IRefreshKernel kernel, View fixedHeader, View fixedFooter);
    void onInitialHeaderAndFooter(int headerHeight, int footerHeight);
    void setScrollBoundaryDecider(ScrollBoundaryDecider boundary);

    void setEnableLoadMoreWhenContentNotFull(boolean enable);

    AnimatorUpdateListener scrollContentWhenFinished(int spinner);
}

File: lib_core/src/main/java/com/eelly/core/widget/refresh/api/IRefreshContent.kt
package com.eelly.core.widget.refresh.api

import android.animation.ValueAnimator.AnimatorUpdateListener
import android.view.MotionEvent
import android.view.View
import android.view.ViewGroup.LayoutParams

/**
 * @author vurtne on 2-Jan-18.
 */
interface IRefreshContent {
    /**
     * @param spinner 微调
     * */
    fun moveSpinner(spinner : Int)
    /**
     * 是否可刷新
     * */
    fun canRefresh(): Boolean
    /**
     * 是否可加载更多
     * */
    fun canLoadMore(): Boolean
    /**
     * 获取宽度
     * */
    fun getMeasuredWidth(): Int
    /**
     * 获取高度
     * */
    fun getMeasuredHeight(): Int
    /**
     * 测量
     * */
    fun measure(widthSpec: Int, heightSpec: Int)
    /**
     * 布局
     * */
    fun layout(left: Int, top: Int, right: Int, bottom: Int)
    /**
     * 获取布局
     * */
    fun getView(): View
    /**
     * 获取可滑动布局
     * */
    fun getScrollableView(): View
    /**
     * 获取Params
     * */
    fun getLayoutParams(): LayoutParams
    /**
     * 触摸到
     * */
    fun onActionDown(e: MotionEvent)
    /**
     * 手指离开或者取消
     * */
    fun onActionUpOrCancel()
    fun fling(velocity: Int)
    fun setUpComponent(kernel: IRefreshKernel, fixedHeader: View, fixedFooter: View)
    fun onInitialHeaderAndFooter(headerHeight: Int, footerHeight: Int)
    fun setScrollBoundaryDecider(boundary: ScrollBoundaryDecider)
    fun setEnableLoadMoreWhenContentNotFull(enable: Boolean)
    fun scrollContentWhenFinished(spinner: Int): AnimatorUpdateListener
}